misc/pstring.cpp

```diff
@@ -38,140 +38,23 @@
  * $NoKeywords: $
  */
 
-#include <string.h>
-#include <stdarg.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include "pstring.h"
-
-#ifdef __LINUX__
-#include "lnxfix.h"
-#endif
+#include <cctype>
+#include <cstring>
 
-//	Pvsprintf
-//		Similar to vsprintf/_vsnprintf, however handles the case of a buffer overflow.  Arguments are the
-//	same as _vsnprintf.  In the case of count, pass in the size of the buffer, in the case where there is
-//	a buffer overflow, a \0 will be tacked on as the last byte in the buffer, ensuring a valid string.
-int Pvsprintf(char *buffer, int count, const char *format, va_list argptr) {
-  int ret = _vsnprintf(buffer, count - 1, format, argptr);
-  if (ret == -1)
-    buffer[count - 1] = '\0';
-  return ret;
-}
+#include <algorithm>
+#include <string_view>
 
-//	Psprintf
-//		Similar to sprintf/_snprintf, however handles the case of a buffer overflow.  Arguments are the
-//	same as _snprintf.  In the case of count, pass in the size of the buffer, in the case where there is
-//	a buffer overflow, a \0 will be tacked on as the last byte in the buffer, ensuring a valid string.
-int Psprintf(char *buffer, int count, const char *format, ...) {
-  va_list ap;
-  va_start(ap, format);
-  int ret = Pvsprintf(buffer, count, format, ap);
-  va_end(ap);
-  return ret;
-}
+#include "pstring.h"
 
 // CleanupStr
-//		this function strips all leading and trailing spaces, keeping internal spaces.  this goes
-//		for tabs too.
-int CleanupStr(char *dest, const char *src, int destlen) {
-  int i, j, err, begin = 0, end = 0, len;
-
-  err = 0;
-
-  len = strlen(src);
-  for (i = 0; i < len; i++) {
-    char ch;
-    ch = src[i];
-
-    //	 mark beginning.
-    if ((ch > ' ' && ch > '\t') && err < 1) {
-      err = 1;
-      begin = i;
-      end = i;
-    } else if (ch == ' ' && err == 1) {
-      err = 2;
-      end = i;
-    } else if (ch > ' ' && err >= 1) {
-      end = i;
-    }
+//    This function strips all leading and trailing spaces, keeping internal spaces. This goes for tabs too.
+int CleanupStr(char *dest, const char *src, std::size_t destlen) {
+  if ((src != nullptr) && (dest != nullptr)) {
+    std::string_view s{src};
+    s.remove_prefix(std::count_if(s.begin(), s.end(), [](unsigned char ch) { return !std::isspace(ch); }));
+    s.remove_suffix(std::count_if(s.rbegin(), s.rend(), [](unsigned char ch) { return !std::isspace(ch); }));
+    std::strncpy(dest, s.data(), destlen);
```

Comment:
This is dangerous move. `string_view.data()` never returns null-terminated string, so at dest's end may be garbage. You had to manually detect end of string and add \0 to end.